pallets/elections/src/lib.rs

```diff
@@ -443,26 +459,46 @@ pub mod pallet {
         type Error = ElectionError;
         type DataProvider = T::DataProvider;
 
-        /// The elections are PoA so only the nodes listed in the Validators will be elected as
-        /// validators.
-        ///
-        /// We calculate the supports for them for the sake of eras payouts.
+        /// We calculate the supports for each validator. The external validators are chosen as:
+        /// 1) "`NextEraNonReservedValidators` that are staking and are not banned" in case of Permissioned ElectionOpenness
+        /// 2) "All staking and not banned validators" in case of Permissionless ElectionOpenness
         fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
-            Self::kick_out_underperformed_non_reserved_validators();
+            Self::emit_fresh_bans_event();
```

Comment:
Ideally, we'd emit only events for genuine bans, as the previous version did for kicks. By genuine I mean here those we are actually removed from `non_reserved` nodes - ie theoretically it's possible there's some bogus validator on the list on which emit will be emitted, but it won't actually be in `staking.validators` map etc

Reply:
In elect, we make sure that validators from given era do stake.